### PR TITLE
Added missing attributes and improvements to deployment yaml file and…

### PR DESCRIPTION
**Pull request by robece**
… normalized yaml file names

- Added features to deployment: nodeSelector, securityContext, tolerations, affinity

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Thanks for your PR, I'll review it today!

File: keda/templates/deployment.yaml [hunk 1]
```diff
@@ -41,30 +49,48 @@ spec:
           - name: grpc-certs
             mountPath: /grpccerts
           {{- end }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
         - name: {{ .Values.operatorName }}-metrics-apiserver
-          image: {{ .Values.image.metricsAdapter }}
+          image: "{{ .Values.image.metricsAdapter }}:{{ .Chart.AppVersion }}"
```
Comment (tomkerkhove) [inline keda/templates/deployment.yaml]: Ok for you if we add this @ahmelsayed? It's fine for me.

Comment (robece) [reply keda/templates/deployment.yaml]: Thanks Tom, I just realized that I added the incorrect ports in the deployment.yaml file, if you like I can add this change in a new commit and pull request or if you can help me to replace it also is good for me:

This is the correct piece that should be in deployment.yaml line 65-71:
```
          ports:
            - containerPort: {{ .Values.service.portHttpsTarget }}
              name: https
              protocol: TCP
            - containerPort: {{ .Values.service.portHttpTarget }}
              name: http
              protocol: TCP
```
Thanks,
RC

Comment (robece) [reply keda/templates/deployment.yaml]: Thanks Tom, I just realized that I added the incorrect ports in the deployment.yaml file, if you like I can add this change in a new commit and pull request or if you can help me to replace it also is good for me.

This is the correct piece that should be in deployment.yaml line 65-71:
```
          ports:
            - containerPort: {{ .Values.service.portHttpsTarget }}
              name: https
              protocol: TCP
            - containerPort: {{ .Values.service.portHttpTarget }}
              name: http
              protocol: TCP
```
Thanks,
RC

Comment (robece) [reply keda/templates/deployment.yaml]: Thanks Tom, I just realized that I added the incorrect ports in the deployment.yaml file, I going to close this PR and send a new one with the changes updated.

**Comment by robece** [comment]
Closed PR because I'm sending a new PR for approval.

**Pull request closed without merging**